### DISCUSSION TITLE
Add support for Podcasts/audiobooks

### DISCUSSION THREAD

**Discussion by dpomnean** [music-assistant/support#429] [Feature requests and ideas]
I would like to see a podcast category, like you can find in the Spotify Integration entity.

**Comment by erkr** [comment]
Moved to discussions, raise your votes!

**Reply by Rjaap** [reply]
I would like too see this also. Btw, great addon!

**Comment by manbeast76** [comment]
I also would like to see this.

**Comment by n8detar** [comment]
Expanding on this, being able to add other sources for podcasts, or podcasts directly would be extremely nice. I know this project is for music, but many of the features, like starting playback again after a notification, would be really nice for podcasts as well.

**Comment by sefininio** [comment]
I'd like to see podcasts as well. Awesome addon!!!

**Comment by TheGroundZero** [comment]
The ability to add a URL to a feed would be nice.
Could maybe be integrated together with the [Feedreader](https://www.home-assistant.io/integrations/feedreader) integration to automatically download latest episode or add its URL to a playlist.

**Comment by cdrfun** [comment]
I don't know if we should separate "Spotify Podcast" and Podcast, maybe @erkr has an opinion on that?
For podcasts, an option to add gpodder.net as source would be awesome. Also, OPML support would be favorable.

**Reply by marcelveldt** [reply]
If I'm going to add podcast/audiobook support it will be implemented as a first class citizen so next to mediatypes artist, album, track, radio and playlist there will be a "podcast" media type which can be provided by providers like spotify but also others.

Strictly taken even 2 new media types as you will need to identify both the podcast itself and its episodes.

**Reply by DutchJaFO** [reply]
Audiobooks would probably need to be an additonal category as well, although in theory they could be treated as Podcasts/episode.

At the very least we would have to disable the 'crossfade' feature whenever podcasts/audiobooks are played as that will make the experience sub-optimal (to say the least).

**Reply by WebfinityDesign** [reply]
Adding gpodder.net in theory would be a very nice feature but gpodder.net is broken most of the time. The server is at the max of the capacity. In August they said they were working on it, but the problem still exists. Looks like they gave up on it

**Reply by fyksen** [reply]
The gpodder service is not great. But the protocol seems solid.
Nextcloud has a addon calles gpodder sync: 
https://github.com/thrillfall/nextcloud-gpodder

Antennapod, kde podcast app and some other podcast apps already supports it. This would be an awesome fit for this project! 🥰

**Reply by Thyraz** [reply]
> If I'm going to add podcast/audiobook support it will be implemented as a first class citizen so next to mediatypes artist, album, track, radio and playlist there will be a "podcast" media type which can be provided by providers like spotify but also others.
> 
> Strictly taken even 2 new media types as you will need to identify both the podcast itself and its episodes.


@marcelveldt  yes, this 2 layer approach is also what I would have in mind.

Based on the pocast / audiobook type it might also be necessary to add entries manually like it's also possible in the radio UI?
For some things like Spotify podcasts it might be possible to add them automatically, as you can follow shows here.

But audiobook series are often simple albums sorted under a single artist.
So adding them by https://open.spotify.com/intl-xx/artist/XYZ would be nice.
Also adding podcasts using classic rss feeds this way.

Saving the current listening position server-side for both, audiobooks and podcasts would be awesome too, so you can continue from any device.
Together with the request to allow the browser as a player target to play audio from Music Assistant e.g. in the car using your phone completes this story. 🙂

**Comment by hillbicks** [comment]
I'm just getting started with music assistant, so first of all thanks for creating this addon!

I'm a bit confused though. Is this discussion solely for creating the category button? Because right now, I can't find the podcasts when I'm searching for them, which is a bit strange to be honest. I would expect it to show all the content that is on spotify. I just searched for a couple of podcasts and couldn't find them.

**Reply by erkr** [reply]
Hi @hillbicks 
Podcasts are not available in MA (yet). Discussions are valuable to poll general interest and discuss how to implement them.
Next major MA release will for example natively support Samba network shares and multiple accounts per music provider (based on a lot of requests)
Best Eric

**Reply by hillbicks** [reply]
@erkr : Thanks for the quick reply. At least now I know that it is not something on my end. 

Multi user support sounds very good as well. 

Again, thanks for the answer and your work on this. Will keep an eye out for these features.

**Reply by erkr** [reply]
> Again, thanks for the answer and your work on this. Will keep an eye out for these features. 

I'm just helping out with moderating issues and testing. Credits are on Marcel, thx

**Comment by WebfinityDesign** [comment]
I would also like to have podcasts/audiobooks in MA. It's nice to have have all the audio in one place and also use it in some automations

**Comment by mattsteg** [comment]
I'd like to endorse this request as well.

**Reply by Dunkelschunkel** [reply]
Are there any updates relating podcasts?
Integration would be fabulous

**Reply by marcelveldt** [reply]
That is not scheduled for the 2.0 release, it will be one of the features that comes after that (e.g. 2.1 release)

**Comment by Dunkelschunkel** [comment]
Thank you

Jul 10, 2023 4:58:05 PM Marcel van der Veldt ***@***.***>:

**Comment by SurreyMichael** [comment]
Would love podcast access.  

I use PocketCasts app as Spotify was really problematic playing podcasts on my Sonos speakers (PocketCasts works really well), so really interested how playing podcasts via music assistant will turn out.

**Reply by orangesunny** [reply]
I would love to see PocketCasts integration! There is no reason to create anything from scratch, as most users probably track their podcasts in some app already. PocketCasts are getting more open since they became part of Automatic, but could be tricky for intergation, though. Apps are open-sourced, but there is currently no API, only Tasker integration on Android. 😢

**Reply by djtecha** [reply]
This would be an awesome feature for my partner. Only thing holding them back.

**Comment by Blackhex** [comment]
Integration of https://podcastaddict.com/app or other podcasts management app would be awesome.

**Comment by Erudition** [comment]
Is there a way to play a podcast given the URL of the media file?

**Reply by OzGav** [reply]
I don’t believe so.

**Reply by marcelveldt** [reply]
Yes, you can add it as a track or radio station manually but it wont be recognized as podcast yet

**Reply by DarkFox** [reply]
If you're using Home Assistant, you can add a Scrape sensor that gets the latest episode. Just add the feed url, and then these options in the sensor config:
```
selector: enclosure:nth-of-type(1)
index: 0
attribute: url
```

Then you can use the sensor state in a mass.play_media service call.

It's not perfect, but it works well enough for my morning news podcasts.

**Reply by Erudition** [reply]
Thanks, I did something similar a few months ago. Good enough for now

**Comment by b-abadie** [comment]
I've been crawling the web for a way to play podcasts (the classic RSS type, not the pseudo-podcast available through a proprietary service) on my Sonos speakers. The closest I've been is the gpodder integration of HA, but gpodder.net is broken (long-term overloaded).

A cool feature of MA would be to add RSS podcasts, automatically or manually downloading episodes, tracking "played" status, etc.
A very cool feature would be to sync all this with a custom gpodder endpoint (I'm thinking https://github.com/thrillfall/nextcloud-gpodder for instance).

Thank you for all your work !

**Reply by LukasDJacobs** [reply]
For the moment I use AntennaPod (OpenSource, Android, great podcast player!) in combination with an app called AirMusic (which does cost money, but it's a relatively cheap one time payment). It works reasonably well so far and I use it daily, but I'm still hoping for native Sonos support on AntennaPod. That would just be a smoother experience. There's an open issue for that, but nobody implemented it yet, so kinda like the situation here

**Comment by Jab2870** [comment]
First off, thanks for all the work that has already gone into this amazing tool.

I would just like to chime in a bit and say that support for audiobook sources and podcast sources would be amazing. All my audiobooks and podcasts are currently in [Audiobookshelf](https://github.com/advplyr/audiobookshelf).

If I could use musicassistant to play audiobooks from there, and for them to stay in sync, I'd be over the moon. Audiobookshelf has pretty comprehensive documentation for it's api, and it's also self hosted so I would think would be a pretty good place to start. 

Regardless, thanks again for all the work you've put into this.

**Reply by lhassell** [reply]
Just here to echo this. As a self-hosted solution, I feel like Audiobookshelf deserves to be included as it fits with the ethos of Home Assistant.

**Comment by dunxd** [comment]
I am excited to see this added in the future. My asks are:
* Support OPML for bringing in subscriptions from another app
* Option to stream podcasts rather than download - but at minimum it should be possible to automatically delete played episodes. I don't want to fill up storage on my HA/MA server with old episodes.
* Some way to sync played status with other apps - I listen to podcasts in my car using Podcast addict. If MA can remove podcasts from the queue that have been completed on another device that would be nice. I have no idea if this is even possible.
* If a partially listened to podcast from elsewhere could pick up playback from the last position, even better!
* Allow skip back and forward by a configurable number of seconds
Maybe I am asking for the moon on a stick with some of these. Being able to handle podcast subscriptions _at all_ would be huge.

**Comment by nmoenks** [comment]
+1 please make this happen! 
I currently have all podcasts in [Audiobookshelf](https://github.com/advplyr/audiobookshelf) as well - which is connected to Plex - which is picked up by MA..

**Reply by DavidOliMar** [reply]
Good afternoon, my name is David, I would be interested in you telling me if it is possible something else about how you have it implemented, I am very interested in this topic because I have been trying to integrate this into HomeAssistant for a long time. I hope that one day they will implement it in MusicAssistant ; )

**Reply by nmoenks** [reply]
Hello David,
i currently have it set up on an Unraid system ([https://unraid.net/](https://unraid.net/)) this way: 

- Plex running in a docker container ([https://www.plex.tv/de/](https://www.plex.tv/de/))
- Audiobookshelf running in docker container ([https://www.audiobookshelf.org/](https://www.audiobookshelf.org/))
- Home Assistant running as a Virtual Machine
- Music Assistant running inside HA

Audiobookshelf just pulls the podcasts and saves them into a specific folder. 
Plex is monitoring this folder and adds the files automatically to the library - which is connected to Music Assistant. 

Its not an ideal setup as the tagging of the files in Audiobookself / Plex is not perfect but it is currently the most comfortable way (i found) to get podcasts into MA. 

I really hope as well, that Audiobookshelf (or something similiar) will be integrated to MA some day :) 
And a way to filter for podcasts would be really nice - for example instead of the "Radio" button in the main menu

**Reply by DavidOliMar** [reply]
Tweedledee good night, sorry for taking so long to answer you but these days I'm on vacation and a little off the computer.

I think it's a very good idea as you have it implemented
I'm going to try to replicate it and test it. 
Maybe I see the problem with the synchronization when listening to podcast on the mobile every day, I don't know how you have that implemented. I have seen that there is an application in Beta for iOS of Audiobookshelf but at the moment it is full and I can't access it. 
I still hope that our request comes to fruition and can be implemented in the future in HomeAssistant

Kind Regards

**Reply by marcelveldt** [reply]
Podcast and Audiobook support will come to MA soon but first we are focusing on stability. We want to have a stable foundation before we start adding new features. So the past releases have all been about stability and speed fixes. Once we feel comfortable that things are in a good place, we have a whole bunch of cool new features waiting to be added. Just a bit of patience is what is needed.

**Reply by DavidOliMar** [reply]
Thank you very much for your answer Marcelveldt, I am very happy to read these words. Today I am very happy with MusicAssistant and it is the music server of all the SonoS in my house. I look forward to all the advances that can be made...

**Comment by seppeel** [comment]
@marcelveldt 
Can you tell if this is targeted for 2.4 or is it planned for a later release?
thanks!

**Reply by marcelveldt** [reply]
If all goes well, 2.4 but that is not a 100% guarantee

**Reply by saeugetier** [reply]
I have experimented with RSS feeds for podcasts.

https://github.com/music-assistant/server/pull/1732

**Comment by quadcom** [comment]
If simply modifying the providers is an easier approach, YTM has podcast subscriptions in it. They just are not listed in MA. Rather than reinventing the wheel how about looking into pulling the data from the providers already configured?

**Reply by marcelveldt** [reply]
the data needs a place to land - that place is not yet there (but will be soon). So the answer remains the same: have a little bit more patience please.

**Reply by hardwareadictos** [reply]
Could be nice to add Audio Book Shelf (a part from audiobooks it has podcasts libraries) and some projects like this one: https://apps.nextcloud.com/apps/gpoddersync

It allows users to self host podcast feed sync on OpenSOurce platforms like Nextcloud or AntennaPod

**Comment by seppeel** [comment]
I'm so excited and i just can't hide it.
https://github.com/music-assistant/server/commit/d00ffbe7685ec8d411301d9474ec8de2e1fa444a

**Comment by balazsmark** [comment]
It would be a game-changer for lot of us.

**Comment by tylerlucas** [comment]
This one would be big! Hoping to see it soon

**Comment by freddeh** [comment]
Yes, this would be so awesome to listen to my morning podcast automatically, preferably by using my Spotify account

**Comment by Lowrider614** [comment]
+1 from my side. Kids would love to hear Maus zum Hören ober the big stereo

**Comment by DaWheelz** [comment]
This would be a great feature as I would be able to play the news podcast in the morning!

**Comment by OzGav** [comment]
Implemented in 2.4.0 beta 9

Now providers are required to be written. This is a separate task.

**Reply by seppeel** [reply]
I tried a bit with the new beta and RSS Feed player provider. Looks good so far!

One thing i noticed is sorting. Every podcast feed should have a "published" metadata field which i would like to use to sort the episodes. the default "position" sorting looks weird in the podcast examples i used.

**Reply by freakshock88** [reply]
I tried the new subsonic podcast integration using Gonic as a subsonic server. I saw that @khers that contributed this implementation also used Gonic. I can see my podcast show up in MA, however when I try to play an episode I get this error log:
```
[31m2025-01-06 21:28:05.835 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/app/venv/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 478, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/aiohttp/web_app.py", line 567, in _handle
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/controllers/metadata.py", line 394, in handle_imageproxy
    image_data = await self.get_thumbnail(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/controllers/metadata.py", line 371, in get_thumbnail
    thumbnail = await get_image_thumb(
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/helpers/images.py", line 68, in get_image_thumb
    img_data = await get_image_data(mass, path_or_url, provider)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/helpers/images.py", line 33, in get_image_data
    if resolved_image := await prov.resolve_image(path_or_url):
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/providers/opensubsonic/sonic_provider.py", line 495, in resolve_image
    return await asyncio.to_thread(_get_cover_art)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/music_assistant/providers/opensubsonic/sonic_provider.py", line 492, in _get_cover_art
    with self._conn.getCoverArt(path) as art:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/libopensonic/connection.py", line 751, in getCoverArt
    self._checkStatus(dres)
  File "/app/venv/lib/python3.12/site-packages/libopensonic/connection.py", line 2508, in _checkStatus
    raise exc(result['error']['message'])
libopensonic.errors.DataNotFoundError: Artwork not found[0m
[31m2025-01-06 21:28:11.240 ERROR (MainThread) [music_assistant.webserver] Error handling message: player_queues/play_media: Episode pd-2$!$pe-141 not found[0m
```

**Reply by saeugetier** [reply]
> I tried a bit with the new beta and RSS Feed player provider. Looks good so far!
> 
> One thing i noticed is sorting. Every podcast feed should have a "published" metadata field which i would like to use to sort the episodes. the default "position" sorting looks weird in the podcast examples i used.

I think @marcelveldt did already some fixes on the ordering.

**Reply by freddeh** [reply]
> Implemented in 2.4.0 beta 9
> 
> Now providers are required to be written. This is a separate task.

so no separate podcast menu, or am I too blind to find it?

**Reply by Thyraz** [reply]
Should be like that:

<img width="313" alt="image" src="https://github.com/user-attachments/assets/2602cced-dfa7-4832-a01f-f0e535b4d6cb" />

**Reply by freddeh** [reply]
> Should be like that:
> 
> <img alt="image" width="313" src="https://private-user-images.githubusercontent.com/170099/400565747-2602cced-dfa7-4832-a01f-f0e535b4d6cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMDIwMjIsIm5iZiI6MTczNjIwMTcyMiwicGF0aCI6Ii8xNzAwOTkvNDAwNTY1NzQ3LTI2MDJjY2VkLWRmYTctNDgzMi1hMDFmLWYwZTUzNWI0ZDZjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNlQyMjE1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iN2UwOWRkYWYwYzVkMzA0ZmRmZmM4MzExNjQxZGJjMzExNDlkYjIyYmY3YTA2ZWE5OTI2YWJmNjRhYzNlYWEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QVY9VonJIPBNqQFia8nDZD_skB0bdrBPUMvnJNA6zSk">

Sorry, my fault. Too stupid to know that if I pull a docker image as :beta, and then call docker run without the :beta that I basically ignore the previously pulled image...

**Reply by diskmemo** [reply]
thanks al lot for the new beta features.
I tried the new podcast rss-feed:
with german podcast feeds i get an explicit - error 
example: https://www.swr.de/~podcast/swr1/rp/meilensteine-podcast-100.xml

**Reply by OzGav** [reply]
@diskmemo if you have a problem then open your own Q&A so it can be looked at more closely

**Reply by saeugetier** [reply]
> thanks al lot for the new beta features. I tried the new podcast rss-feed: with german podcast feeds i get an explicit - error example: https://www.swr.de/~podcast/swr1/rp/meilensteine-podcast-100.xml

I will take a look.

**Reply by saeugetier** [reply]
Fix has been pushed: https://github.com/music-assistant/server/pull/1832

**Reply by khers** [reply]
> I tried the new subsonic podcast integration using Gonic as a subsonic server. I saw that @khers that contributed this implementation also used Gonic. I can see my podcast show up in MA, however when I try to play an episode I get this error log:
> 
> ```
> �[31m2025-01-06 21:28:05.835 ERROR (MainThread) [aiohttp.server] Error handling request
> Traceback (most recent call last):
>   File "/app/venv/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 478, in _handle_request
>     resp = await request_handler(request)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/aiohttp/web_app.py", line 567, in _handle
>     return await handler(request)
>            ^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/controllers/metadata.py", line 394, in handle_imageproxy
>     image_data = await self.get_thumbnail(
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/controllers/metadata.py", line 371, in get_thumbnail
>     thumbnail = await get_image_thumb(
>                 ^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/helpers/images.py", line 68, in get_image_thumb
>     img_data = await get_image_data(mass, path_or_url, provider)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/helpers/images.py", line 33, in get_image_data
>     if resolved_image := await prov.resolve_image(path_or_url):
>                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/providers/opensubsonic/sonic_provider.py", line 495, in resolve_image
>     return await asyncio.to_thread(_get_cover_art)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/asyncio/threads.py", line 25, in to_thread
>     return await loop.run_in_executor(None, func_call)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/music_assistant/providers/opensubsonic/sonic_provider.py", line 492, in _get_cover_art
>     with self._conn.getCoverArt(path) as art:
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/app/venv/lib/python3.12/site-packages/libopensonic/connection.py", line 751, in getCoverArt
>     self._checkStatus(dres)
>   File "/app/venv/lib/python3.12/site-packages/libopensonic/connection.py", line 2508, in _checkStatus
>     raise exc(result['error']['message'])
> libopensonic.errors.DataNotFoundError: Artwork not found�[0m
> �[31m2025-01-06 21:28:11.240 ERROR (MainThread) [music_assistant.webserver] Error handling message: player_queues/play_media: Episode pd-2$!$pe-141 not found�[0m
> ```

Does this podcast have a cover image associated with it?

**Reply by freakshock88** [reply]
Yes, it does
![Screenshot_20250107_202141_Home Assistant](https://github.com/user-attachments/assets/3bdd88df-2433-4876-a7d5-1cbc22f2a548)
At least, the podcast itself does. The episode perhaps does not?

**Reply by DaWheelz** [reply]
I can't see the Podcast section, where can I find it in the Beta?

**Reply by OzGav** [reply]
Go to the MA SETTINGS and FRONTEND and add the view. You then need to add a provider that supports what you use. More providers are getting added every update at the moment! Audible and audiobookshelf appear to be progressing ok but no promises until they are released.

**Reply by DaWheelz** [reply]
Found it, thanks! Really looking forward to how this will work out in future updates.

**Reply by seppeel** [reply]
> I tried a bit with the new beta and RSS Feed player provider. Looks good so far!
> 
> One thing i noticed is sorting. Every podcast feed should have a "published" metadata field which i would like to use to sort the episodes. the default "position" sorting looks weird in the podcast examples i used.

I would like to renew my request to sort with "published date". "Position" is often messed up in my podcasts.

**Comment by Pittermaennchen** [comment]
> Implemented in 2.4.0 beta 9
> 
> Now providers are required to be written. This is a separate task.

Can anyone add audiobookshelf, please? Would do it myself, if I knew how.

**Comment by DavidOliMar** [comment]
Good afternoon. 
First of all, thank you for the effort in the implementation of the podcasts in MusicAsistant. I have installed the beta and I have been testing a little the news about it. 
The problem that I currently have and I don't know how to solve is the synchronization of the exact point where I have stopped listening to the pending podcasts when I get home. 
With Pocketcast for example, which is where I usually listen to it, I have real synchronization and when I get home I continue right where I left them but the tests I have done with Gonic or Audiobookshelf I can't do anything similar about it. 
How have you solved that issue in relation to MusicAssistant or is there something about it that will be implemented soon? 
Thank you very much again in advance.
Greetings ; )

**Reply by khers** [reply]
This is an outstanding problem. I will be implementing bookmarking for podcasts using the open subsonic api, but I haven't done it yet. Thanks for testing the Subsonic implementation!

**Reply by DavidOliMar** [reply]
Perfect, many to you for the work in being able to implement it.

I look forward to hearing from you.

**Comment by cerw** [comment]
Silly Q but how can i install beta and can downgrade back ? ta

**Reply by OzGav** [reply]
https://www.music-assistant.io/release/#running-parallel-server-versions

**Comment by DavidOliMar** [comment]
Hello 👋🏻
Today I have updated and I have been able to add Audiobookshelf as a podcast provider.
Is there any way to integrate the playlists we have in Audiobookshelf?
And although I have been with the Audiobookshelf server for a short time, does anyone know if it is possible to synchronize the progress of the Podcasts or Audiobook to continue at the same point when you get home and play the content in MusicAssistant?
Thank you very much in advance for your answers ; )

**Reply by marcelveldt** [reply]
Synchronizing play state is still WIP - keep an eye out on the beta updates, also new providers are being added almost every beta update.

**Reply by DavidOliMar** [reply]
Ok, I'll be attentive with a lot of desire
Thanks a lot 😀

**Comment by cerw** [comment]
Got it installed and it can see my podcast, but it wont play it? Wont see any error in console... how to report bug ?

```
2025-01-19 11:31:06.127 ERROR (MainThread) [music_assistant.audio.media_stream] Error while streaming podcastfeed--CMsW5jmX://podcast_episode/7fa3640e-f7b0-52e6-abf0-75a70f423071: No audio was received
2025-01-19 11:31:06.134 INFO (MainThread) [music_assistant.streams] Start Streaming queue track: podcastfeed--CMsW5jmX://podcast_episode/4f6b0b31-bd29-592d-8769-49e11154a829 (Novinky na alternativní scéně | Josef Sedloň | Jan 16) for queue Bathroom speaker

```

**Reply by saeugetier** [reply]
Please report a bug in https://github.com/music-assistant/support

In order to be able to reproduce your problem, please give information about your feed url.

**Comment by WhiteSockedDancer** [comment]
> Synchronizing play state is still WIP - keep an eye out on the beta updates, also new providers are being added almost every beta update.

May i also suggest adding a way to get the play-state for different users (maybe by adding the same provider with different credentials).

My use case would be an household where everybody can listen to his own potcasts/audiobooks in parallel. Probably even the same media with different playstates synced to the different (audiobookshelf-)users.

I would love to help and participate in any way. Feel free to contact! :)

**Reply by WhiteSockedDancer** [reply]
Getting the right user on a play-prompt could be difficult.. hmm..

**Reply by marcelveldt** [reply]
MA is not multi user aware at all and will not be for the upcoming future. There are way more important feature requests that have priority first. I think chances are low that multiple users in one household listen to the same podcast/audiobook so the current behavior where you can resume any podcast or audiobook will work for 99% of the cases just fine. If you need something else, use Home Assistant and the power of scripts and automations to completely go wild. All MA actions and states are available in HA

**Reply by WhiteSockedDancer** [reply]
I'm totally fine with having this no high priority or getting it implemented at all. 

But since this is a discussion about Audiobooks/Features in MA:

I think having people listening to the same podcast in the same household is a quite commen usecase. My wife and i listen to the same audiobooks and podcasts in different states (some books i already listened to some years ago, others we listen in parallel on our daily routes and so on..).
The most important part is, that the actual progress is synced with ABS. If my wife starts an audiobook it should resume at her state synced from ABS. When i start an audiobook it should start at the state of my account.

I guess it is maybe possible with scripts and automations in HA by asking ABS via API for the progress and then let MA start the book at the specific timestamp. 

Or did i misinterpreted "play state" as "progress" but you meant "playing" "paused"?

**Reply by marcelveldt** [reply]
I added something for this on the backlog, just as an FYI. It will not have priority soon but so you are aware something will come eventually. https://github.com/orgs/music-assistant/projects/2/views/1?pane=issue&itemId=98497460

**Reply by WhiteSockedDancer** [reply]
> I added something for this on the backlog, just as an FYI. It will not have priority soon but so you are aware something will come eventually. https://github.com/orgs/music-assistant/projects/2/views/1?pane=issue&itemId=98497460

Sounds great. I'm looking forward to it. :)

**Comment by agdamsbo** [comment]
Just spun up the latest beta version (20). Really great to have podcasts integrated. I was able to add a simple RSS feed and playback was flawless. I use Antennapod on my phone and sync/backup with the Nextcloud gPodder implementation. I see gpodder mentioned before. I dont have the skills to implement it myself, so just voting for implementing it as a provider. I see a python library exists (https://github.com/gpodder/mygpoclient) and there are different options for selfhosting (Nextcloud and others) as alternatives to the gpodder.net, which is a little underpowered, I think. Very happy to use Music Assistant daily around the house with several speakers/boards. Thank you!

**Comment by khers** [comment]
I agree and am working on support for this in the core but life has me focussing elsewhere at the moment. I can't promise an ETA but it will get there.

**Comment by p3ck** [comment]
The podcast import fails when I try and import http://rss.art19.com/the-daily

**Comment by freddeh** [comment]
Is there anyone planning to add Spotify Podcast API support? It seems my favorite podcasts are not available via Podcast RSS, which is a pity.

**Reply by OzGav** [reply]
Eventually. The devs are busy with other things at the moment,

**Comment by freddeh** [comment]
Is it already possible to use one of the actions from HA to play a specific episode of a podcast without knowing the episode GUID?

**Comment by bastgau** [comment]
Thanks !!

**Comment by cerw** [comment]
This works great! but sometimes can fastword podcast? any idea?

**Reply by OzGav** [reply]
You will need to provide a lot more detail for us to provide an answer.

**Comment by tylerlucas** [comment]
Is there a bookmark feature for audiobooks? I see that a feature [request was opened for bookmarks here](https://github.com/orgs/music-assistant/discussions/416), but was closed due to being a duplicate with this request.

**Reply by pedroangelini** [reply]
Would be very cool to have! I started using the audiobooks feature yesterday, and did not realize it was just a few months old. It feels so polished already!

Couple of things I'm missing are:
- when in audiobook or podcast mode, the back/forward control should change to "jump x seconds ahead/backwards", instead of jump to next track
- also for audiobooks, the time position horizontal bar become pretty much unusable because the audio is so long - ideally it should show your position in the chapter, not in the whole book.

**Reply by SurreyMichael** [reply]
Just in case helpful, I have been using a script for skipping ahead/behind for several years written with help from the HA community.  It works when I am playing a podcast via HA/sonos/pocketcasts assume it would work for podcasts in MA:

alias: skip
sequence:
  - data_template:
      entity_id: " {{ entity }} "
      seek_position: " {{ ( as_timestamp(now()) - as_timestamp(state_attr(entity, 'media_position_updated_at')) + state_attr(entity, 'media_position') ) | float + interval }} "
    action: media_player.media_seek

Just give a media_player as entity and interval (e.g. +60 or -30) when you invoke.

To me podcasts in MA is only useful if it works with a podcast client I can use on my iPhone (such as pocketcasts).  The point of podcasts is that you listen to them when and where you want - not just at home - and pick up where you left off.

**Reply by Thyraz** [reply]
MA could be that client in the future. ;) 
The local audio output already allows to use it on your phone (and then e.g. in the car because of that) as long as you can reach it from outside your home network.

I use VPN with an auto-connect profile on my iPhone, which automatically connects if any app/webpage tries to reach a service in my home domain.

Podcasts might need some more time to evolve to a first-class solution over dedicated podcasts apps, as this is a pretty new feature.
But for music I already use MA instead of the Spotify app when I'm on the go.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
